test/unit/core/store/middleware.ts

```diff
@@ -0,0 +1,93 @@
+import { ActionCreators } from 'redux-undo';
+import * as uuid from 'uuid';
+import Actions from '../../../../src/core/actions';
+import * as Events from '../../../../src/core/events';
+import createMiddleware, { errorHandler, idGenerator } from '../../../../src/core/store/middleware';
+import suite from '../../_suite';
+
+suite('store middleware', ({ expect, spy, stub }) => {
+
+  describe('createMiddleware()', () => {
+    it('should return middleware', () => {
+      const middleware1 = { c: 'd' };
+      const middleware2 = { e: 'f' };
+      const creator1 = spy(() => middleware1);
+      const creator2 = spy(() => middleware2);
+      const flux: any = { a: 'b' };
+
+      const middleware = createMiddleware([creator1, creator2], flux);
+
+      expect(creator1).to.be.calledWith(flux);
+      expect(creator2).to.be.calledWith(flux);
+      expect(middleware).to.eql([middleware1, middleware2]);
+    });
+  });
+
+  describe('idGenerator()', () => {
+    it('should add id if action type is in whitelist', (done) => {
+      const id = '1234';
+      const idKey = 'myId';
+      const v1 = stub(uuid, 'v1').returns(id);
+      const whitelist = ['a', 'b', 'c'];
+
+      idGenerator(idKey, whitelist)(null)(null)((action) => {
+        expect(action).to.eql({ type: 'b', c: 'd', meta: { [idKey]: id } });
+        done();
```

Comment:
Shouldn't need `done()`.  Make `next` a spy and expect it to be called with the object.